apps/profiles/views.py

```diff
@@ -572,10 +572,13 @@ def list(self, request, format=None):
         serializer = PrivacySerializer(privacy)
         return response.Response(serializer.data)
 
-    def update(self, request, pk=None):
+    def put(self, request, pk=None):
         privacy = Privacy.objects.get(user=request.user)
         serializer = PrivacySerializer(privacy, data=request.data)
-        return response.Response(serializer.data)
+        if serializer.is_valid():
```

Comment:
If you use add the kwarg `raise_exception=True` here it will automatically return a HTTP 400 BAD REQUEST if the serializer validation fails, rather than a 200 OK for an invalid/non-chaning PUT request.

https://www.django-rest-framework.org/api-guide/serializers/#raising-an-exception-on-invalid-data

Reply:
Ah, I did not know that.
Should I still return something if it is not valid, or just count on the exception handling in DRF to take care of it in all cases?

Reply:
It'll raise an exception so you won't need to return anything, no